### PR TITLE
[AI-1190] Implement and test augmentations

### PR DESCRIPTION
**Pull request by ChristofferEdlund**
# Problem
We want to introduce the possibility to use Albumentation transforms with darwin-py torch datasets. 

# Solution
Introducing an `AlbumentationsTransform` class in torch.transforms that can be be used in the following manner:

```
from darwin.torch.dataset import (
    ClassificationDataset,
    InstanceSegmentationDataset,
    ObjectDetectionDataset,
    SemanticSegmentationDataset,
)

from darwin.torch.transforms import AlbumentationsTransform

dataset_path = Path(r"/path/to/local/darwin/dataset")
transform = AlbumentationsTransform.from_path('/tmp/transform.json')

inst_dataset = InstanceSegmentationDataset(dataset_path=dataset_path, transform=transform)
```
One can initilize the `AlbumentationsTransform` in three ways:
- AlbumentationsTransform(albumentation_transform) <- needs an albumentations transformation
- AlbumentationsTransform.frompath(path) <- a path pointing to a .json or .yaml file defining the transformation
- AlbumentationsTransform.fromdict(dict) <- a dictionary defining the transform

To read more about the dictionary and file formats supported, we refer to the [albumentations documentation.](https://albumentations.ai/docs/examples/serialization/#Serializing-an-augmentation-pipeline-to-a-JSON-or-YAML-file)

Further.
- Instance segmentation dataset output is changed for bounding boxes to be `coco` format (X, Y, W, H) to be consistent with darwin-json annotations and the ObjectDetecion torch dataset. [BREAKING CHANGE]
- Clamping is introduced to ObjectDetection bboxes that is outside of image for more robust data loading.

# Changelog
- Introduced albumentation transform support for darwin torch datasts
- [BREAKING CHANGE] darwin.torch.dataset.InstanceSegmentationDataset has bbox coordinates changes from `pascal_voc` to `coco` format (X, Y, H, W)
- darwin.torch.dataset.ObjectDetectionDataset clamps bbox coordinates out-of-bound.

### CODE REVIEW

File: darwin/torch/transforms.py
Comment (dorfmanrobert) [inline darwin/torch/transforms.py:200]: can remove commented out lines

**Comment by owencjones** [comment]
Updated title so that it's easier for me on deployment 😉

**Review by almazan — APPROVED** [review]
All changes look good to me.

**Review by almazan — CHANGES_REQUESTED** [review]
Some comments need to be addressed before approval

File: darwin/torch/transforms.py
Comment (almazan) [inline darwin/torch/transforms.py:217]: I believe this logic was added for cases with bboxes of either width or height of size 0px. Should we add it back? The easiest way to add it back is to compute `keep` using a `tmp_boxes` converted to `x,y,x,y` using `tmp_boxes[:, 2:] += boxes[:, :2]` and then filter `boxes` using `keep`.

**Review by almazan — CHANGES_REQUESTED** [review]

File: darwin/torch/transforms.py
Comment (almazan) [inline darwin/torch/transforms.py:202]: This is not clamping bboxes anymore, since now indexes 2 and 3 are height and width instead of x2 and y2. Boxes now could go outside of the image.

**Review by almazan — APPROVED** [review]

**Pull request merged**
